ms2/src/org/labkey/ms2/compare/CompareQuery.java

```diff
@@ -153,7 +157,12 @@ protected void generateSql(BindException errors)
 
     protected void addColumn(String name)
     {
-        _columns.add(FieldKey.fromParts(name));
+        // Fake up a column map so we can use Sort.getOrderByClause()
+        FieldKey fk = FieldKey.fromParts(name);
+        BaseColumnInfo col = new BaseColumnInfo(fk, JdbcType.VARCHAR);
+        col.setSortFieldKeys(Collections.singletonList(fk));
+        col.setAlias(MS2Manager.getSqlDialect().makeDatabaseIdentifier(name));

```

Comment:
Since names are mixed-case, we just have to make sure our usages are consistent.  But it would be obvious if there were a problem.